src/commands/helm-deploy.yml

```diff
@@ -0,0 +1,98 @@
+description: >
+  Deploys Helm to ShuttleCrawler, assumes image version of component is set in .version
+parameters:
+  app-name:
+    type: string
+    description: Name of application to be deployed
+  environment:
+    type: string
+    description: Environment to be deployed on (prod, staging, dev)
+  component:
+    type: string
+    description: Name of folder which contains the Helm Chart
+  domain:
+    type: string
+    description: Domain (K8s namespace) to deploy to
+  path:
+    type: string
+    description: Path to .chart folder of application
+  short-region:
+    type: string
+    default: usc1
+    description: Short-region to deploy Compute Engine resources
+  region:
+    type: string
+    default: us-central1
+    description: Region to deploy Compute Engine resources
+  registry-region:
+    type: string
+    default: us-central1
+    description: Region of Artifact Registry
+  dry-run:
+    type: boolean
+    default: false
+    description: If enabled, simulate an upgrade
+  channel:
+    type: string
+    description: Slack channel to send notifications to
+    default: deploys-staging
+steps:
+- checkout
+- gcp-authorize:
+    environment-variable-name: GCLOUD_SERVICE_ACCOUNT
+    do-helm-login: 'true'
+- run:
+    name: Setting up ShuttleCrawler Google and K8 context
+    command: |
+      gcloud container clusters get-credentials \
+        platform-<< parameters.environment >>-<< parameters.region >>-cluster \
+        --zone << parameters.region >> \
+        --project ${GCLOUD_PROJECT_ID}
+- run:
+    name: Preparing Helm upgrade parameters
+    command: |
+      params=(--atomic --install --cleanup-on-fail --debug)
+      [[ -f << parameters.path >>/.chart/<< parameters.component >>/<< parameters.environment >>.yaml ]] && params+=("-f << parameters.environment >>.yaml")
+      params+=("-f values.yaml")
+      params+=("--namespace=<< parameters.domain >>")
+      params+=("--set global.environment=<< parameters.environment >>")
+      params+=("--set global.projectId=${GCLOUD_PROJECT_ID}")
+      params+=("--set global.shortRegion=<< parameters.short-region >>")
+      params+=("--set global.region=<< parameters.region >>")
+      params+=("--set global.registryRegion=<< parameters.registry-region >>")
```

Comment:
Ohh I like it, because hopefully we only have one registry (even though those are region specific GCP resources), even though we may host services on multiple regions